test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/Search/CompositeSearchTestFixture.cs

```diff
@@ -26,32 +26,56 @@ public class CompositeSearchTestFixture : HttpIntegrationTestFixture<Startup>
             "ObservationWithBloodPressure",
         };
 
+        private static readonly string[] DocumentReferenceTestFiles =
+        {
+            "DocumentReference-example",
+            "DocumentReference-example-002",
+            "DocumentReference-example-003",
+        };
+
         public CompositeSearchTestFixture(DataStore dataStore, Format format)
             : base(dataStore, format)
         {
-            var resultDictionary = new Dictionary<string, Observation>(TestFileNames.Length);
+            Observations = CreateResultDictionary<Observation>(ObservationTestFileNames);
+            DocumentReferences = CreateResultDictionary<DocumentReference>(DocumentReferenceTestFiles);
+        }
+
+        public string TestSessionId { get; } = Guid.NewGuid().ToString();
+
+        public IReadOnlyDictionary<string, Observation> Observations { get; }
 
-            for (int i = 0; i < TestFileNames.Length; i++)
+        public IReadOnlyDictionary<string, DocumentReference> DocumentReferences { get; }
+
+        private Dictionary<string, T> CreateResultDictionary<T>(string[] files)
+            where T : Resource
+        {
+            var resultDictionary = new Dictionary<string, T>(files.Length);
+
+            for (int i = 0; i < files.Length; i++)
             {
-                string testFileName = TestFileNames[i];
+                string testFileName = files[i];
 
-                Observation result = FhirClient.CreateResourcesAsync<Observation>(() =>
+                T result = FhirClient.CreateResourcesAsync<T>(() =>
                 {
-                    Observation observation = Samples.GetJsonSample<Observation>(testFileName);
+                    T observation = Samples.GetJsonSample<T>(testFileName);
```

Comment:
nit: this should be resource not observation since it could be observation or document reference